src/mrsimulator/spin_system/__init__.py

```diff
@@ -402,6 +402,28 @@ def parse_dict_with_units(cls, py_dict: dict):
 
         return super().parse_dict_with_units(py_dict_copy)
 
+    def rotate(self, euler_angles):
```

Comment:
Add type hints. `def rotate(self, euler_angles: list):` Not clear if the argument is a list or a tuple.